src/Servers.jl

```diff
@@ -315,12 +316,24 @@ function listen(f::Function,
         tcpref[] = tcpserver
     end
 
+    if host isa Sockets.InetAddr # build debugging info
+        hostname = string(host.host)
+        hostport = string(host.port)
+    else
+        hostname = string(host)
+        hostport = ""
+    end
+    listenloop(f, tcpserver, sslconfig, hostname, hostport, pipeline_limit, require_ssl_verification, tcpisvalid, connectioncounter; kw...)
+end
```

Comment:
Why is `listenloop` split into a seperate function?
It seems a bit cumbersome to have to pass all those positional parameters across in the right order.
Will `listenloop` be called from somewhere else too?

Reply:
The split was to improve type inference; once the higher-level `listen` function figures out the exact types of `tcpserver`, `sslconfig`, and `tcpisvalid`, it can pass them all to `listenloop` which can then compile solely for the types provided; previously, type inference struggled quite a bit with all the uncertainty.

Reply:
Fair enough. Was this causing a noticeable performance problem?

Reply:
Yes, it was a 5-10% performance hit from my benchmarks.